hoomd/mpcd/ConfinedStreamingMethod.h

```diff
@@ -129,12 +136,15 @@ void ConfinedStreamingMethod<Geometry>::stream(unsigned int timestep)
         // propagate the particle to its new position ballistically
         Scalar dt_remain = m_mpcd_dt;
         bool collide = true;
+        bool bounced = false;
         do
             {
             pos += dt_remain * vel;
             collide = m_geom->detectCollision(pos, vel, dt_remain);
+            bounced |= collide;
             }
         while (dt_remain > 0 && collide);
+        h_bounced.data[cur_p] = bounced;
```

Comment:
Again, group things logically for readability. This should go below with lines 158 & 159 as a write operation per particle.